### PR TITLE
Close #11948: Support hide edit suggestion button for search suggestions

### LINKED ISSUE #11948: History and bookmark search suggestions should not support edit suggetion
Currently history and bookmark suggestions have a downward arrow that is meant for autocomplete suggestions.  Remove these from the history and bookmark result

Screenshot_20220331_172252
s

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProvider.kt [hunk 1]
```diff
@@ -71,7 +71,6 @@ class BookmarksStorageSuggestionProvider(
                 flags = setOf(AwesomeBar.Suggestion.Flag.BOOKMARK),
                 title = result.title,
                 description = result.url,
-                editSuggestion = result.url,
```
Comment (csadilek) [inline components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProvider.kt:74]: I don't think it's right to remove this for the `Awesomebar` in general. Maybe you want to be able to configure this so it can behave different for bookmark and history search?

Comment (csadilek) [reply components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProvider.kt:74]: One easy option is to add a parameter to the provider to configure whether or not to add edit suggestions.

Comment (rocketsroger) [reply components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProvider.kt:74]: Sure, I'll make it configurable so we can turn it on or off.  Thanks

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt [hunk 2]
```diff
@@ -37,7 +38,8 @@ fun AwesomeBar(
     onSuggestionClicked: (AwesomeBar.Suggestion) -> Unit,
     onAutoComplete: (AwesomeBar.Suggestion) -> Unit,
     onScroll: () -> Unit = {},
-    profiler: Profiler? = null
+    profiler: Profiler? = null,
+    showEditSuggestions: () -> Boolean = { true },
```
Comment (rocketsroger) [inline components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:42]: I chose to use a function here so we can update this dynamically when we're ready with unified search.

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt [hunk 3]
```diff
@@ -37,6 +38,7 @@ internal fun Suggestions(
         modifier = Modifier.testTag("mozac.awesomebar.suggestions")
     ) {
         val suggestions = fetcher.state.value
+        val showEditSuggestion: Boolean = showEditSuggestions()
```
Comment (rocketsroger) [inline components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:41]: Added this so we don't call the function every time below.

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt (hunk 2 above)
Comment (csadilek) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:42]: Can you elaborate why a function makes sense here? I don't see why this should be decided dynamically each time Suggestions are drawn. In other words, why would this change between queries of the same `AwesomeBar` instance?

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt (hunk 3 above)
Comment (csadilek) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:41]: (See my comment above. Definitely good to not call it for every suggestion, but I don't even see why we need to call it dynamically on a Suggestion level. Maybe I am missing a feature you're thinking of.)

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt (hunk 2 above)
Comment (rocketsroger) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:42]: I could move it up higher.  But I was thinking about future requirement for unified search. When the user switch over the history search, we should dynamically stop showing the button.

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt (hunk 3 above)
Comment (rocketsroger) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:41]: In this case I'm not calling for every suggestion.  I added this so every suggestion will use the same value set on this line.

Comment (csadilek) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:41]: I know, like I wrote above, but you're calling it for every set of suggestions which seems unnecessary.

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt (hunk 2 above)
Comment (csadilek) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:42]: What do you mean by "When the user switch over the history search"? Happy to talk through next week, the must be a requirement I am missing :).

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt (hunk 3 above)
Comment (gabrielluong) [inline components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/internal/Suggestions.kt:41]: Driveby: Can we remove the explicit `: Boolean`

File: components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt (hunk 2 above)
Comment (rocketsroger) [reply components/compose/awesomebar/src/main/java/mozilla/components/compose/browser/awesomebar/AwesomeBar.kt:42]: Sure, let's talk.  It could be that I don't need to pass the function down to the `Suggestions` level.  Thanks

File: components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/HistoryMetadataSuggestionProviderTest.kt [hunk 4]
```diff
@@ -170,4 +171,18 @@ class HistoryMetadataSuggestionProviderTest {
             emittedFacts.first()
         )
     }
+
+    @Test
+    fun `WHEN provider is created with showEditSuggstion set to false THEN editSuggestion is set to null`() = runBlocking {
```
Comment (gabrielluong) [inline components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/HistoryMetadataSuggestionProviderTest.kt:176]:
```suggestion
    fun `WHEN provider is created with showEditSuggestion set to false THEN editSuggestion is set to null`() = runBlocking {
```

File: components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/CombinedHistorySuggestionProviderTest.kt [hunk 5]
```diff
@@ -158,4 +159,19 @@ class CombinedHistorySuggestionProviderTest {
         assertEquals("https://www.mozilla.com/pocket", result[2].description)
         assertEquals(2, result[2].score)
     }
+
+    @Test
+    fun `WHEN provider is created with showEditSuggstion set to false THEN editSuggestion is set to null`() = runBlocking {
```
Comment (gabrielluong) [inline components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/CombinedHistorySuggestionProviderTest.kt:164]:
```suggestion
    fun `WHEN provider is created with showEditSuggestion set to false THEN editSuggestion is set to null`() = runBlocking {
```

File: components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProviderTest.kt [hunk 6]
```diff
@@ -105,6 +106,24 @@ class BookmarksStorageSuggestionProviderTest {
         verify(engine, times(1)).speculativeConnect(eq(suggestions[0].description!!))
     }
 
+    @Test
+    fun `WHEN provider is created with showEditSuggstion set to false THEN editSuggestion is set to null`() = runBlocking {
```
Comment (gabrielluong) [inline components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/BookmarksStorageSuggestionProviderTest.kt:110]:
```suggestion
    fun `WHEN provider is created with showEditSuggestion set to false THEN editSuggestion is set to null`() = runBlocking {
```

**Review by csadilek — APPROVED** [review]
🚢

File: components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/HistoryStorageSuggestionProviderTest.kt [hunk 7]
```diff
@@ -56,6 +57,18 @@ class HistoryStorageSuggestionProviderTest {
         assertEquals("http://www.mozilla.com/", suggestions[0].description)
     }
 
+    @Test
+    fun `WHEN provider is created with showEditSuggstion set to false THEN editSuggestion is set to null`() = runBlocking {
```
[the remark this answers is not in the document]
Comment (rocketsroger) [reply components/feature/awesomebar/src/test/java/mozilla/components/feature/awesomebar/provider/HistoryStorageSuggestionProviderTest.kt:61]: Sure, I'll rename without using parameter name
